presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayContains.java

```diff
@@ -13,120 +13,51 @@
-import static com.facebook.presto.metadata.Signature.internalOperator;
+import static com.facebook.presto.metadata.OperatorType.EQUAL;
 import static com.facebook.presto.spi.StandardErrorCode.INTERNAL_ERROR;
-import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.facebook.presto.spi.type.TypeUtils.readNativeValue;
 
+@ScalarFunction("contains")
+@Description("Determines whether given value exists in the array")
 public final class ArrayContains
-        extends SqlScalarFunction
 {
-    public static final ArrayContains ARRAY_CONTAINS = new ArrayContains();
-    private static final String FUNCTION_NAME = "contains";
-    private static final MethodHandle METHOD_HANDLE_UNKNOWN = methodHandle(ArrayContains.class, "arrayWithUnknownType", Type.class, MethodHandle.class, Block.class, Void.class);
-
-    public ArrayContains()
-    {
-        super(FUNCTION_NAME, ImmutableList.of(comparableTypeParameter("T")), ImmutableList.of(), StandardTypes.BOOLEAN, ImmutableList.of("array(T)", "T"));
-    }
-
-    @Override
-    public boolean isHidden()
-    {
-        return false;
-    }
-
-    @Override
-    public boolean isDeterministic()
-    {
-        return true;
-    }
-
-    @Override
-    public String getDescription()
-    {
-        return "Determines whether given value exists in the array";
-    }
-
-    @Override
-    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
-    {
-        Type type = boundVariables.getTypeVariable("T");
-
-        MethodHandle methodHandle;
-        MethodHandle equalsHandle = functionRegistry.getScalarFunctionImplementation(internalOperator(OperatorType.EQUAL, BooleanType.BOOLEAN, ImmutableList.of(type, type))).getMethodHandle();
-
-        List<Boolean> nullableArguments;
-        if (type.getJavaType() == void.class) {
-            nullableArguments = ImmutableList.of(false, true);
-            methodHandle = METHOD_HANDLE_UNKNOWN;
-        }
-        else {
-            nullableArguments = ImmutableList.of(false, false);
-            methodHandle = methodHandle(ArrayContains.class, "contains", Type.class, MethodHandle.class, Block.class, type.getJavaType());
-        }
-
-        return new ScalarFunctionImplementation(true, nullableArguments, methodHandle.bindTo(type).bindTo(equalsHandle), isDeterministic());
-    }
-
-    public static Boolean arrayWithUnknownType(Type elementType, MethodHandle equals, Block arrayBlock, Void value)
-    {
-        return null;
-    }
-
-    public static Boolean contains(Type elementType, MethodHandle equals, Block arrayBlock, Block value)
+    private ArrayContains() {}
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    @Nullable
+    public static Boolean contains(
+            @OperatorDependency(operator = EQUAL, returnType = StandardTypes.BOOLEAN, argumentTypes = {"T", "T"}) MethodHandle equalsFunction,
```

Comment:
Can you change all the parameter names back to the previous ones? It's very hard to review this because everything shows that it was changed